js/tabs.js

```diff
@@ -322,6 +309,54 @@ Tabs.prototype.openFiles = function() {
       this.openFileEntry.bind(this));
 };
 
+/**
+ * @param {function()} callback
+ * Invoke the save dialog for all tabs with unsaved progress. Does not close any tabs.
+ */
+Tabs.prototype.promptAllUnsaved = function(callback) {
+  this.promptAllUnsavedFromIndex_(0, callback);
+}
+
+Tabs.prototype.promptAllUnsavedFromIndex_ = function(i, callback) {
+  if (this.tabs_.length > i) {
+    var tab = this.tabs_[i];
+    if (tab.isSaved()) {
+      this.promptAllUnsavedFromIndex_(i + 1, callback);
+    } else {
+      this.showTab(this.tabs_[i].getId());
+      this.promptSave_(tab, function(answer) {
+        if (answer === 'yes') {
+          this.save(tab, false /* close */);
+        } else if (answer === 'cancel') {
+          return;
+        }
+        this.promptAllUnsavedFromIndex_(i + 1, callback);
+      }.bind(this));
+    }
+  }
+  else {
```

Comment:
I would suggest...

if (i >= this.tabs_.length) {
  callback();
  return;
}

Which will unindent the other block and make it simpler to read.

Reply:
Done.